jis-plugins/src/main/java/pl/edu/uj/JImageStream/filters/Histogram.java

```diff
@@ -0,0 +1,245 @@
+package pl.edu.uj.JImageStream.filters;
+
+import java.awt.Color;
+import java.awt.image.BufferedImage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import pl.edu.uj.JImageStream.api.core.Filter;
+import pl.edu.uj.JImageStream.model.ColorChannel;
+import pl.edu.uj.JImageStream.model.Pixel;
+
+public class Histogram extends Filter{
+
+    private int[][] histogram = new int[4][256]; // 0=R, 1=G, 2=B, 3=A
+    private int[] blackAndWhiteHistogram = new int[256];
+    private boolean[] channel = new boolean[4]; // 0=R, 1=G, 2=B, 3=A
+    private ColorChannel[] channels;
+    private final int TOPMARGIN = 0;
+    private BufferedImage image = null;
+    private BufferedImage output = null;
+    private int outputWidth;
+    private int outputHeight;
+    private boolean readyToApply;
+    private boolean blackAndWhite;
+
+    public Histogram(BufferedImage image, ColorChannel ... channels){
+        readyToApply = false;
+        blackAndWhite = false;
+        this.image = image;
+        this.channels = channels;
+    }
+
+    private void init(){
+        readyToApply = true;
+        for(int j = 0 ; j < 256 ; ++j){
+            blackAndWhiteHistogram[ j ] = 0;
+            for(int i = 0 ; i < 4 ; ++i){
+                histogram[ i ][ j ] = 0;
+            }
+        }
+        for(int i = 0 ; i < 4 ; ++i){
+            channel[ i ] = false;
+        }
+
+        output = new BufferedImage(getSourceWidth(), getSourceHeight(), BufferedImage.TYPE_INT_ARGB);
+        outputHeight = output.getHeight();
+        outputWidth = output.getWidth();
+
+        if( channels.length > 0 ){
+            setChannels(channels);
+        } else {
+            blackAndWhite = true;
+        }
+        if(( outputWidth >= 256 )&&( image != null )){
+            calculateHistogram();
+            setScale();
+            drawHistogram();
+        } else {
+            Logger logger = LogManager.getLogger(this.getClass());
+            logger.warn("Streamable Image Width should be at least 256 pixels");
+        }
+    }
+
+    private void calculateHistogram(){
```

Comment:
i feel like this should be divided into tho methods one for colored histogram, one for black and white.